File: call-clj/src/main/java/capclug/demo/callclj/Nine.java
/*
 * $Id$
 * Copyright (C) 2010 ARIN
 */
package capclug.demo.callclj;

import capclug.demo.cljfromjava.nine;

/**
 * @version $Rev$
 */
public class Nine
{
    public static void main( String[] args )
    {
        nine nine = new nine();
        nine.foo( "42d" ); // requires a String!

        nine.bar( "43d" );
    }
}

File: clj-from-java/src/main/clojure/capclug/demo/cljfromjava/nine.clj
(ns capclug.demo.cljfromjava.nine
  (:gen-class
    :methods [[foo [String] void]
              [bar [String] void]]))

(defn -foo [this #^Object baz]
  (println "will I work? " baz))

(defn -bar [this #^Object baz]
  (println "will I work? " (.toUpperCase baz)))
